api/lib/infrastructure/repositories/session-repository.js

```diff
@@ -47,26 +49,20 @@ module.exports = {
   },
 
   async getWithCertificationCandidates(idSession) {
-    try {
-      const session = await BookshelfSession
-        .where({ id: idSession })
-        .fetch({ withRelated: [
-          {
-            'certificationCandidates': function(qb) {
-              qb.select(Bookshelf.knex.raw('*'));
-              qb.orderByRaw('LOWER("certification-candidates"."lastName") asc');
-              qb.orderByRaw('LOWER("certification-candidates"."firstName") asc');
-            },
-          },
-        ],
-        });
-      return bookshelfToDomainConverter.buildDomainObject(BookshelfSession, session);
-    } catch (err) {
-      if (err instanceof BookshelfSession.NotFoundError) {
-        throw new NotFoundError('La session n\'existe pas ou son accès est restreint');
-      }
-      throw err;
+
+    const session = await knex('sessions')
```

Comment:
En une requete, ca donne: 

```suggestion
        const results = await knex.select('sessions.*').select({
      certificationCandidates: knex.raw(`
        json_agg("certification-candidates".* order by lower("lastName"), lower("firstName"))
        `,
      ),
    }).from('sessions')
      .join('certification-candidates', 'certification-candidates.sessionId', 'sessions.id')
      .groupBy('sessions.id')
      .where({ 'sessions.id': idSession })
      .first();

    if (!results) {
      throw new NotFoundError('La session n\'existe pas ou son accès est restreint');
    }

    return _toDomain(results);
```

 avec pour le _toDomains

```javascript
function _toDomain(results) {
  const toDomainCertificationCandidates = results.certificationCandidates.map((candidate) => {
    return new CertificationCandidate({ ...candidate });
  });

  return new Session({
    ...results,
    certificationCandidates: toDomainCertificationCandidates,
  });
```

Reply:
@aceol ça fait péter les test d'acceptance :cry:

Reply:
effectivement, il faut mettre un leftJoin au lieu d'un join (innerJoin)